### PR TITLE
Store archived timestamp and persist it

### LINKED ISSUE #733: Store archived timestamp
Add an optional `archived` property to each Child. It should be set to `timestamp()` when a thought is archived and persisted across all modifications like `created`.

### PR DESCRIPTION
**Pull request by shresthabijay**
fixes #733

### CODE REVIEW

**Comment by shresthabijay** [comment]
@raineorshine I have added `archived` field to individual thought inside `thoughtIndex`. It will be created only when thoughts are being archived and it persists on thought change and thought move. Please review the changes. Thanks!

File: src/reducers/existingThoughtMove.js [hunk 1]
```diff
@@ -40,13 +40,17 @@ export default (state, { oldPath, newPath, offset }) => {
   const sameContext = equalArrays(oldContext, newContext)
   const oldThought = getThought(state, value)
 
-  /** Find exact thought from thoughtIndex. */
-  const exactThought = () => oldThought.contexts.find(thought => equalArrays(thought.context, oldContext) && thought.rank === oldRank)
+  const isArchived = newThoughts[newThoughts.length - 2] === '=archive'
```
Comment (raineorshine) [inline src/reducers/existingThoughtMove.js:43]: Recommend to use the existing `util/isArchived`.

Comment (shresthabijay) [reply src/reducers/existingThoughtMove.js:43]: `isArchived` takes `Lexeme` as parameter but we only have access to `newPath` in `exisitingThoughtMove`.

Comment (raineorshine) [reply src/reducers/existingThoughtMove.js:43]: Of course, my bad.

Regardless, a thought is archived if any of its ancestors is `=archive`, not just the penultimate components of its path.

```js
newThoughts.indexOf('=archive') !== -1
```

**Pull request merged**